lib/puppet/provider/foreman_smartproxy/rest_v3.rb

```diff
@@ -4,7 +4,15 @@
   def proxy
     @proxy ||= begin
       r = request(:get, 'api/v2/smart_proxies', :search => %{name="#{resource[:name]}"})
-      raise Puppet::Error.new("Proxy #{resource[:name]} cannot be retrieved: #{error_message(r)}") unless success?(r)
+      #raise Puppet::Error.new("Proxy #{resource[:name]} cannot be retrieved: #{error_message(r)}") unless success?(r)
+      unless success?(r)
+        case error_message(r)
+        when /response 401/
+          raise Puppet::Error.new("401 RESPONSE PLACEHOLDER ERROR MESSAGE")
+        when /response 503/
+          raise Puppet::Error.new("503 RESPONSE PLACEHOLDER ERROR MESSAGE")
```

Comment:
Please see the API for https://www.rubydoc.info/stdlib/net/Net/HTTPResponse. There's no need to parse the error message if you have the code.

You can also take a look at [error_message(response)](https://github.com/theforeman/puppet-foreman/blob/master/lib/puppet/provider/foreman_resource/rest_v3.rb#L97-L99) which may be a much better place to handle it.